python/ray/_private/test_utils.py

```diff
@@ -377,6 +377,34 @@ def wait_for_condition(
     raise RuntimeError(message)
 
 
+async def async_wait_for_condition(
```

Comment:
Nice!  Surprised we never needed a util function like this before

Reply:
me too.... also, I really wish there was a good way to share code between sync & async code in python. it's sad I needed to copy-paste this